Tests/Units/Consumer/ConsumerManager.php

```diff
@@ -0,0 +1,142 @@
+ *
+ * A class to test the consumer manager
+ */
+class ConsumerManager extends test
+{
+    /**
+     * @return void
+     */
+    public function testShouldConsumeMessages()
+    {
+        $eventDispatcherMock = $this->getEventDispatcherMock();
+        $this
+            ->given(
+                $consumer = new Base(),
+                $consumer->setConsumer($this->getRdKafkaConsumerMock()),
+                $consumer->addTopic('name'),
+                $consumer->setEventDispatcher($eventDispatcherMock)
+            )
+            ->if($result = $consumer->consume())
+            ->then
+                ->array($result)
+                    ->isEmpty()
+                ->array($topicsStates = $consumer->getTopicsConsumptionState())
+                ->mock($eventDispatcherMock)
+                    ->call('dispatch')
+                        ->withArguments('kafka.event', new Event('consumer'))
+                            ->once()
+            ;
+    }
+
+    /**
+     * @return void
+     */
+    public function testShouldConsumeMessagesWithDataAlreadyExisten()
+    {
+        $eventDispatcherMock = $this->getEventDispatcherMock();
+        $this
+            ->given(
+                $consumer = new Base(),
+                $consumer->setConsumer($this->getRdKafkaConsumerMock()),
+                $consumer->addTopic('testName'),
+                $consumer->setEventDispatcher($eventDispatcherMock)
+            )
+            ->if($result = $consumer->consume(
+                ['testName' => [
+                        0 => 1
+                    ]
+                ])
+            )
+            ->then
+                ->array($result)
+                    ->isEmpty()
+                ->array($topicsStates = $consumer->getTopicsConsumptionState())
+                ->mock($eventDispatcherMock)
+                    ->call('dispatch')
+                        ->withArguments('kafka.event', new Event('consumer'))
+                            ->once()
+        ;
+    }
+
+    /**
+     * @return void
+     */
+    public function testShouldNoSendAnEventIfNotSetWhenConsumingMessages()
+    {
+        $eventDispatcherMock = $this->getEventDispatcherMock();
+        $this
+            ->given(
+                $consumer = new Base(),
+                $consumer->setConsumer($this->getRdKafkaConsumerMock()),
+                $consumer->addTopic('name')
+            )
+            ->if($result = $consumer->consume())
+            ->then
+                ->mock($eventDispatcherMock)
+                    ->call('dispatch')
+                        ->never()
+        ;
+    }
+
+    /**
+     * @return \mock\Symfony\Component\EventDispatcher\EventDispatcher
+     */
+    protected function getEventDispatcherMock()
+    {
+        $this->mockGenerator->orphanize('__construct');
+        $this->getMockGenerator()->shuntParentClassCalls();
+        $mock = new \mock\Symfony\Component\EventDispatcher\EventDispatcher();
```

Comment:
If your mock must do nothing, you could just mock the [`EventDispatcherInterface`](https://github.com/symfony/event-dispatcher/blob/master/EventDispatcherInterface.php) parent interface.